src/com/facebook/buck/swift/SwiftRuntimeNativeLinkable.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2016-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.facebook.buck.swift;
+
+import static com.facebook.buck.swift.SwiftDescriptions.isSharedRequested;
+
+import com.facebook.buck.apple.AppleCxxPlatform;
+import com.facebook.buck.cxx.CxxPlatform;
+import com.facebook.buck.cxx.Linker;
+import com.facebook.buck.cxx.NativeLinkable;
+import com.facebook.buck.cxx.NativeLinkableInput;
+import com.facebook.buck.model.BuildTarget;
+import com.facebook.buck.model.Flavor;
+import com.facebook.buck.model.ImmutableFlavor;
+import com.facebook.buck.parser.NoSuchBuildTargetException;
+import com.facebook.buck.rules.SourcePath;
+import com.facebook.buck.rules.args.StringArg;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import java.nio.file.Path;
+
+/**
+ * Pseudo linkable for representing Swift runtime library's linker arguments.
+ */
+final class SwiftRuntimeNativeLinkable implements NativeLinkable {
```

Comment:
I think the problem with NativeLinkableInput is that it is concatenated in order (that is why we are using `ImmutableList` in [here](https://github.com/facebook/buck/blob/1e1d14f23616505309e4349bdc83867c75379672/src/com/facebook/buck/cxx/AbstractNativeLinkableInput.java#L62-L62)).

Using pseudo `NativeLinkable` works, since all `NativeLinkable` created in `SwiftLibrary#getNativeLinkableDeps` has the same build target, therefore you will have only one `SwiftRuntimeNativeLinkable` instance in the list of accumulated linkables after this: https://github.com/facebook/buck/blob/b83ae5538a7c553ccd10f06b2ee034dccfa412d6/src/com/facebook/buck/cxx/NativeLinkables.java#L231-L238. 

I've already tested. It works pretty well. There could be still a duplication because of this https://github.com/facebook/buck/blob/34fedbea306a4b02bd47d1e5d8847a6bac87f780/src/com/facebook/buck/swift/SwiftLibraryDescription.java#L287-L287. But there can only be a duplication of 2 at most.